### ISSUE TITLE
ignore a part of the page

### ISSUE THREAD

**Issue by ctf0** [turbolinks/turbolinks#155]
currently i have a soundcloud widget, and am trying to get it to load only once and ignore its container on any page redirect through TL , so is it possible ?

- another issue is getting a repeated msg for the widget reloading
![test](https://cloud.githubusercontent.com/assets/7388088/17274610/c32fe824-56e9-11e6-96dd-69e9e81ad1cb.jpg)

**Comment by sstephenson** [comment]
Please see [Persisting Elements Across Page Loads](https://github.com/turbolinks/turbolinks#persisting-elements-across-page-loads) in the documentation.

**Comment by ctf0** [comment]
@sstephenson i have already reviewed it and what it does is that TL will compare that element between visits but it will replace it anyway because its located in the body tag, but what am after is to ignore that part of the body, so TL will treat it like it doesnt even exist.

i tried to edit the `event.data.newBody` but i don't know how to find that specific element and remove it from the body before it replace the old content.

**Comment by grosser** [comment]
FYI workaround in https://github.com/turbolinks/turbolinks/issues/43

**Comment by ctf0** [comment]
@grosser so basically we put anything we dont want to be re-rendered into
```html
<div id="permanent">...stuff...</div>
```
and use
```js
$(document).on('turbolinks:click',function (){
	def turbolink_permanent(id, &block)
		anchor = content_tag(:div, "permanent #{id}", "data-turbolinks-permanent": true, id: id)
		return anchor if request.env["HTTP_TURBOLINKS_REFERRER"]
		html = capture(&block)
		anchor << javascript_tag("document.getElementById('#{id}').innerHTML = #{html.inspect};")
	end
}
```
???

sorry for the noob question but not sure how to use it.

**Comment by grosser** [comment]
it's a erb helper, so you'd do:

```Erb
<%= turbolink_permanent :header do %>
  <header>
<% end %>
```

**Comment by ctf0** [comment]
is it possible to add a version with plain html ?

**Comment by grosser** [comment]
no, this relies on server rendering logic.

you can use `data-turbolinks-permanent=true` to have the same element no re-render ... and then modify it via js ... but if the server response is different it will replace that element
